lib/lambda/update/updatePackage.ts

```diff
@@ -0,0 +1,232 @@
+  updatedId: string;
+}) => {
+  const topicName = process.env.topicName as string;
+  if (!topicName) {
+    throw new Error("Topic name is not defined");
+  }
+  // ID and changeMade are excluded; the rest of the object has to be spread into the new package
+  const {
+    id: _id,
+    changeMade: _changeMade,
+    origin: _origin,
+    ...remainingFields
+  } = currentPackage._source;
+
+  if (!updatedId) {
+    return response({
+      statusCode: 400,
+      body: { message: "New ID required to update package" },
+    });
+  }
+
+  // check if a package with this new ID already exists
+  const packageExists = await getPackage(updatedId);
+  if (packageExists) {
+    return response({
+      statusCode: 400,
+      body: { message: "This ID already exists" },
+    });
+  }
+  // use event of current package to determine how ID should be formatted
+  const packageEvent = await getPackageType(currentPackage._id);
+  const packageSubmissionTypeSchema = events[packageEvent as keyof typeof events].baseSchema;
+
+  if (!packageSubmissionTypeSchema) {
+    return response({
+      statusCode: 500,
+      body: { message: "Could not validate the ID of this type of package." },
+    });
+  }
+
+  const idSchema = packageSubmissionTypeSchema.shape.id;
+  const parsedId = idSchema.safeParse(updatedId);
+
+  if (!parsedId.success) {
+    return response({
+      statusCode: 400,
+      body: parsedId.error.message,
+    });
+  }
+
+  await sendDeleteMessage(currentPackage._id);
+  await produceMessage(
+    topicName,
+    updatedId,
+    JSON.stringify({
+      id: updatedId,
+      idToBeUpdated: currentPackage._id,
+      ...remainingFields,
+      origin: "OneMAC",
+      changeMade: "ID has been updated.",
+      isAdminChange: true,
+      adminChangeType: "update-id",
+    }),
+  );
+
+  return response({
+    statusCode: 200,
+    body: { message: `The ID of package ${currentPackage._id} has been updated to ${updatedId}.` },
+  });
+};
+
+const updatePackageEventBodySchema = z.object({
+  packageId: z.string(),
+  action: z.enum(["update-values", "update-id", "delete"]),
+  updatedId: z.string().optional(),
+  updatedFields: z.record(z.unknown()).optional(),
+  changeReason: z.string().optional(),
+});
+
+export const handler = async (event: APIGatewayEvent) => {
+  if (!event.body) {
+    return response({
+      statusCode: 400,
+      body: { message: "Event body required" },
+    });
+  }
+  try {
+    const parseEventBody = (body: unknown) => {
+      return updatePackageEventBodySchema.parse(typeof body === "string" ? JSON.parse(body) : body);
+    };
+
+    const {
+      packageId,
+      action,
+      updatedId = packageId,
+      updatedFields = {},
+      changeReason,
+    } = parseEventBody(event.body);
+
+    if (!packageId || !action) {
+      return response({
+        statusCode: 400,
+        body: { message: "Package ID and action are required" },
+      });
+    }
+
+    const packageResult = await getPackage(packageId);
```

Comment:
Do you wanna rename this `currentPackage` so it fits with the object property args of `sendUpdateIdMessage` etc?